singer-io/tap-pendo
# TDL-17163: Reduce requested data for pages

# Description of change
[TDL-17163](https://jira.talendforge.org/browse/TDL-17163): Reduce requested data for pages
- Currently, the tap is requesting all the data for the below streams and then writing records which are greater than a bookmark so it might cause an OOM error.
  - Accounts, Features, TrackTypes, Guides, Pages
- Pendo is not supporting any pagination directly but found one filter operator so we added a filter parameter in the request to ask for data greater than bookmark for the below streams. Reference: [ filter operator](https://developers.pendo.io/docs/?bash#operators)
  - accounts, poll_events, guide_events, guides, visitors
- visitor_history is traversing for all the parents currently so optimize it by traversing on updated visitors only.
- Made guide_events independent from a parent stream guide.
 
NOTE: Currently, page_events, feature_events, and track_events are traversing on all the parents so no way to apply date_window on it as we can not fix any starting point to traverse parents. Implement date_windowing or pagination for the same after confirmation.



# Manual QA steps
 - Run sync mode for the above-mentioned streams and verified that the same number of records are emitted compared to the old tap. Also logged length of request output to verify that we are requesting less data now.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch
